ts/screens/messages/MessagesHomeScreen.tsx

```diff
@@ -55,48 +98,87 @@ class MessagesHomeScreen extends React.Component<Props, never> {
-            <MessagesInbox
-              messagesState={lexicallyOrderedMessagesState}
-              servicesById={servicesById}
-              paymentByRptId={paymentsByRptId}
-              onRefresh={refreshMessages}
-              setMessagesArchivedState={updateMessagesArchivedState}
-              navigateToMessageDetail={navigateToMessageDetail}
-            />
-          </Tab>
-          <Tab heading={I18n.t("messages.tab.deadlines")}>
-            <MessagesDeadlines
-              messagesState={lexicallyOrderedMessagesState}
-              onRefresh={refreshMessages}
-              navigateToMessageDetail={navigateToMessageDetail}
-            />
-          </Tab>
-          <Tab heading={I18n.t("messages.tab.archive")}>
-            <MessagesArchive
-              messagesState={lexicallyOrderedMessagesState}
-              servicesById={servicesById}
-              paymentByRptId={paymentsByRptId}
-              onRefresh={refreshMessages}
-              setMessagesArchivedState={updateMessagesArchivedState}
-              navigateToMessageDetail={navigateToMessageDetail}
-            />
-          </Tab>
-        </Tabs>
-      </TopScreenComponent>
+        <Tab heading={I18n.t("messages.tab.inbox")}>
+          <MessagesInbox
+            messagesState={lexicallyOrderedMessagesState}
+            servicesById={servicesById}
+            paymentByRptId={paymentsByRptId}
+            onRefresh={refreshMessages}
+            setMessagesArchivedState={updateMessagesArchivedState}
+            navigateToMessageDetail={navigateToMessageDetail}
+          />
+        </Tab>
+        <Tab heading={I18n.t("messages.tab.deadlines")}>
+          <MessagesDeadlines
+            messagesState={lexicallyOrderedMessagesState}
+            onRefresh={refreshMessages}
+            navigateToMessageDetail={navigateToMessageDetail}
+          />
+        </Tab>
+        <Tab heading={I18n.t("messages.tab.archive")}>
+          <MessagesArchive
+            messagesState={lexicallyOrderedMessagesState}
+            servicesById={servicesById}
+            paymentByRptId={paymentsByRptId}
+            onRefresh={refreshMessages}
+            setMessagesArchivedState={updateMessagesArchivedState}
+            navigateToMessageDetail={navigateToMessageDetail}
+          />
+        </Tab>
+      </Tabs>
     );
-  }
+  };
+
+  /**
+   * Render MessageSearch component.
+   */
+  private renderSearch = () => {
+    const {
+      lexicallyOrderedMessagesState,
+      servicesById,
+      paymentsByRptId,
+      refreshMessages,
+      navigateToMessageDetail
+    } = this.props;
+
+    const { searchText } = this.state;
+
+    return (
+      <MessagesSearch
+        messagesState={lexicallyOrderedMessagesState}
+        servicesById={servicesById}
+        paymentByRptId={paymentsByRptId}
+        onRefresh={refreshMessages}
+        navigateToMessageDetail={navigateToMessageDetail}
+        searchText={searchText.getOrElse("")}
```

Comment:
if `searchText` is empty, why render this component at all?

Reply:
also, can we have a loading indicator somewhere until the search results are ready?